src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js

```diff
@@ -5,39 +5,71 @@ import 'whatwg-fetch';
 
 import './style.scss';
 
-function handleDownload(url, filename, accessToken) {
-  const headers = {};
+class DownloadButton extends React.Component {
+  constructor(props) {
+    super(props);
 
-  if (accessToken) {
-    headers.Authorization = `Bearer ${accessToken}`;
+    this.downloadLink = React.createRef();
+    this.handleDownload = this.handleDownload.bind(this);
   }
 
-  fetch(url, {
-    method: 'GET',
-    headers,
-    responseType: 'blob'
-  }).then((response) => response.blob())
-    .then((blob) => {
-      const link = document.querySelector('.download-button__link');
-      link.href = window.URL.createObjectURL(blob);
-      link.click();
-    });
-}
+  handleDownload(url, filename, accessToken) {
+    const headers = {};
+
+    if (accessToken) {
+      headers.Authorization = `Bearer ${accessToken}`;
+    }
+
+    fetch(url, {
+      method: 'GET',
+      headers,
+      responseType: 'blob'
+    }).then((response) => response.blob())
+        .then((blob) => {
+          if (navigator.msSaveOrOpenBlob) {
+            navigator.msSaveOrOpenBlob(blob, filename);
+          } else {
+            const link = this.downloadLink.current;
+            link.href = window.URL.createObjectURL(blob);
```

Comment:
See the [section on memory management](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL#Memory_management) in the URL docs on MDN; you should revoke the object URL.
See example here: https://developer.mozilla.org/en-US/docs/Web/API/File/Using_files_from_web_applications#Example_Using_object_URLs_to_display_images

Reply:
ik heb nu een cleanere opzet en de revoke erin